src/graphics/texture.rs

```diff
@@ -320,8 +332,6 @@ impl Texture {
         unsafe {
             glPixelStorei(GL_UNPACK_ALIGNMENT, 1); // miniquad always uses row alignment of 1
 
-            Self::apply_swizzle_parameter(self.format);
```

Comment:
I can't really find the answer in the specs and also I did not test it right now, but I feel like swizzle might be reset by glPixelStorei? I remember there was some reason to re-apply both PixelStore and the swizzle on each texture update

Reply:
Source https://registry.khronos.org/OpenGL/specs/es/3.0/es_spec_3.0.pdf

Pixel storage affects the following TexImage commands.

> Pixel storage modes affect the operation of TexImage*D, TexSubImage*D, and
> ReadPixels when one of these commands is issued.

Swizzling is only happening during texture access in the fragment shader.

> When a texture lookup is performed in a fragment shader ...  followed
> by swizzling the components of Cb, controlled by the values of the texture pa-
> rameters TEXTURE_SWIZZLE_R, TEXTURE_SWIZZLE_G, TEXTURE_SWIZZLE_B,
> and TEXTURE_SWIZZLE_A.

Swizzle was first used with the introduction of Alpha texture here 6b867f8c75e1b1fe07776d59bfe12aa6b2f59d91
I think it is safe.

Should I still apply swizzle everytime or write an example? Unfortunately my own project [vidrs](https://github.com/payload/vidrs) I use for testing only works on mac right now.

Reply:
In this branch I changed the quad example to call resize every second. Looks good in firefox and on mac gl.
https://github.com/payload/miniquad/tree/fix-swizzle-example